Project1/GridMap.cpp

```diff
@@ -33,116 +61,46 @@ const Party& GridMap::getParty(const Unit & unit) const {
 	return unit.getParty()->getParty(Passkey<GridMap>());
 }
 
-//Non- const findShortestPaths()//
+//Non- const getShortestPathsMap()//
 /////////////////////////////////////////////////////////////////////////
 
-PathMap GridMap::findShortestPaths(ID start) {
-	return PathMap(*this, getShortestPaths(start, INT_MAX, MobilitySet(MobilityType::values::GROUNDED), false));
-}
-
-PathMap GridMap::findShortestPaths(size_t x, size_t y) {
-	return PathMap(*this, getShortestPaths((*this)[x][y]._id, INT_MAX, MobilitySet(MobilityType::values::GROUNDED), false));
-}
-
-PathMap GridMap::findShortestPaths(ID start, int max_move, MobilitySet mobility) {
+PathMap GridMap::getShortestPathsMap(ID start, int max_move, MobilitySet mobility) {
 	bool intangible = mobility[MobilityType::values::PROJECTILE] || mobility[MobilityType::values::ETHEREAL];
-	return PathMap(*this, getShortestPaths(start, max_move, mobility, intangible));
+	return getShortestPathsMap(start, max_move, mobility, intangible);
 }
 
-PathMap GridMap::findShortestPaths(ID start, int max_move, MobilitySet mobility, bool intangible) {
-	return PathMap(*this, getShortestPaths(start, max_move, mobility, intangible));
+PathMap GridMap::getShortestPathsMap(ID start, int max_move, MobilitySet mobility, bool intangible) {
+	return PathMap( *this, getShortestPathsHelper(start, max_move, mobility, [intangible](const Unit* unit) { return intangible || unit == nullptr; }, *this));
```

Comment:
You can avoid having to pass `*this` as an argument to getShortestPathsHelper by capturing it in this lambda:
`[intangible, *this](const CellEdge& edge) { return intangible || getUnit(edge.)id) == nullptr; }`
and in the other lambda:
`[&unit, *this](const CellEdge& edge) { return unit.getMobility().canPass(getUnit(edge.)id)); }`

then you can just call `canPass(edge)` in the helper

Reply:
getShortestPathsMap(...) needs access to the grid though...